components/automate-cli/cmd/chef-automate/cmdUtils.go

```diff
@@ -0,0 +1,428 @@
+	RemoteJobs(command string, inputFiles map[string]string, outputFiles []string, remoteService string, errorCheck bool, newSSHUtil SSHUtil, resultChan chan CmdResult)
+}
+
+type CmdUtilImpl struct {
+	NodeMap *NodeTypeAndCmd
+	SshUtil SSHUtil
+}
+
+func NewCmdUtil(nodeMap *NodeTypeAndCmd) CmdUtil {
+	return &CmdUtilImpl{
+		NodeMap: nodeMap,
+		SshUtil: NewSSHUtil(&SSHConfig{}),
+	}
+}
+
+func (c *CmdUtilImpl) RunCommand() error {
+	var err error
+	timestamp := time.Now().Format("20060102150405")
+
+	sshConfig := getSshDetails(c.NodeMap.Infra)
+	c.SshUtil.setSSHConfig(sshConfig)
+
+	switch true {
+	case c.NodeMap.Frontend.CmdInputs.NodeType:
+		const remoteService string = CONST_FRONTEND
+		nodeIps, err := preCmdExecCheck(c.NodeMap.Frontend, c.SshUtil, c.NodeMap.Infra, remoteService, timestamp, writer)
+		if err != nil {
+			return err
+		}
+
+		err = c.CmdExecOnNodes(c.NodeMap.Frontend.CmdInputs, nodeIps, remoteService, timestamp, writer)
+	case c.NodeMap.Automate.CmdInputs.NodeType:
+		const remoteService string = CONST_AUTOMATE
+		nodeIps, err := preCmdExecCheck(c.NodeMap.Automate, c.SshUtil, c.NodeMap.Infra, remoteService, timestamp, writer)
+		if err != nil {
+			return err
+		}
+
+		err = c.CmdExecOnNodes(c.NodeMap.Automate.CmdInputs, nodeIps, remoteService, timestamp, writer)
+	case c.NodeMap.Chef_server.CmdInputs.NodeType:
+		const remoteService string = CONST_CHEF_SERVER
+		nodeIps, err := preCmdExecCheck(c.NodeMap.Chef_server, c.SshUtil, c.NodeMap.Infra, remoteService, timestamp, writer)
+		if err != nil {
+			return err
+		}
+
+		err = c.CmdExecOnNodes(c.NodeMap.Chef_server.CmdInputs, nodeIps, remoteService, timestamp, writer)
+	case c.NodeMap.Postgresql.CmdInputs.NodeType:
+		const remoteService string = CONST_POSTGRESQL
+		nodeIps, err := preCmdExecCheck(c.NodeMap.Postgresql, c.SshUtil, c.NodeMap.Infra, remoteService, timestamp, writer)
+		if err != nil {
+			return err
+		}
+
+		err = c.CmdExecOnNodes(c.NodeMap.Postgresql.CmdInputs, nodeIps, remoteService, timestamp, writer)
+	case c.NodeMap.Opensearch.CmdInputs.NodeType:
+		const remoteService string = CONST_OPENSEARCH
+		nodeIps, err := preCmdExecCheck(c.NodeMap.Opensearch, c.SshUtil, c.NodeMap.Infra, remoteService, timestamp, writer)
+		if err != nil {
+			return err
+		}
+
+		err = c.CmdExecOnNodes(c.NodeMap.Opensearch.CmdInputs, nodeIps, remoteService, timestamp, writer)
+	default:
+		return errors.New("Missing or Unsupported flag")
+	}
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *CmdUtilImpl) CmdExecOnNodes(input *CmdInputs, nodeIps []string, remoteService string, timestamp string, cliWriter *cli.Writer) error {
+	resultChan := make(chan CmdResult, len(nodeIps))
+	originalSSHConfig := c.SshUtil.getSSHConfig()
+	inputFiles := input.InputFiles
+	outputFiles := input.Outputfiles
+	command := input.Cmd
+
+	inputFileToOutputFileMap := map[string]string{}
+	for _, file := range inputFiles {
+		configFile := remoteService + "_" + timestamp + "_" + file
```

Comment:
I think you should rename `configFile` variable as this function is generic and will be used for other commands too.

Reply:
Done